packages/core/src/core.js

```
@@ -0,0 +1,225 @@
+      });
+    } else {
+      var blockchain = new Blockchain(opts.networkName, opts.netIDs, opts.version);
+      blockchains.push(blockchain.toJSON());
+    }
+
+    this.configuration = new Configuration(opts.devID, blockchains, opts.maxNodes || 5, opts.requestTimeOut || 10000);
+
+  }
+
+  createWallet(subnetwork, data) {
+    throw new Error("Must implement Create Wallet")
+  }
+  importWallet(address, privateKey, subnetwork, data) {
+    throw new Error("Must implement Import Wallet")
+  }
+
+  // Create a Relay instance
+  createRelay(blockchain, netID, version, data, devID) {
+    try {
+      return new Relay(blockchain, netID, version, data, devID, this.configuration);
+    } catch (error) {
+      throw error;
+    }
+  }
+
+  // Create a Report instance
+  createReport(ip, message) {
+    try {
+      return new Report(ip, message, this.configuration);
+    } catch (error) {
+      throw error;
+    }
+  }
+
+  // Get a Dispatch instance or create one
+  getDispatch() {
+    if (this.dispatch == null) {
+      this.dispatch = new Dispatch(this.configuration);
+    }
+    return this.dispatch;
+  }
+
+  // Filter nodes by netID and blockchain name
+  getNode(netID, network, version) {
+    try {
+      if (this.configuration.nodesIsEmpty()) {
+        throw new Error("Failed to filter nodes. List is empty.");
+      }
+
+      var nodes = []
+      this.configuration.nodes.forEach(element => {
+        if (element.netID == netID && element.network == network && element.version == version) {
+          nodes.push(element);
+        }
+      });
+      if (nodes.length <= 0) {
+        return null
+      }
+      return nodes[Math.floor(Math.random() * nodes.length)];
+    } catch (error) {
+      throw new Error("Failed to filter nodes with error: " + error);
+    }
+  }
+  // Send a report
+  async sendReport(report, callback) {
+    try {
+      // Check for report
+      if (report == null) {
+        throw new Error("Report is null");
```

Wouldn't it better to return an error like you're doing in the other function? Also call the callback with the error.